kubernetes/scheduler/controller_test.go

```diff
@@ -0,0 +1,367 @@
+// Copyright © 2018 Genome Research Limited
+// Author: Theo Barber-Bany
+// <tb15@sanger.ac.uk>.
+//
+//  This file is part of wr.
+//
+//  wr is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  wr is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with wr. If not, see <http://www.gnu.org/licenses/>.
+
+package scheduler_test
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"testing"
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/VertebrateResequencing/wr/cloud"
+	"github.com/VertebrateResequencing/wr/kubernetes/client"
+	kubescheduler "github.com/VertebrateResequencing/wr/kubernetes/scheduler"
+	"github.com/docker/docker/pkg/namesgenerator"
+	"github.com/inconshreveable/log15"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeinformers "k8s.io/client-go/informers"
+)
+
+var lc *client.Kubernetesp
+var autherr error
+var nsErr error
+var testingNamespace string
+var callBackChan chan string
+var badCallBackChan chan *cloud.Server
+var reqChan chan *kubescheduler.Request
+var podAliveChan chan *kubescheduler.PodAlive
+var clientset kubernetes.Interface
+var restConfig *rest.Config
+var skip bool
+
+func init() {
+
+	lc = &client.Kubernetesp{}
+
+	clientset, restConfig, autherr = lc.Authenticate()
+	if autherr != nil {
+		skip = true
+		return
+	}
+
+	rand.Seed(time.Now().UnixNano())
+	testingNamespace = strings.Replace(namesgenerator.GetRandomName(1), "_", "-", -1) + "-wr-testing"
+
+	nsErr = lc.CreateNewNamespace(testingNamespace)
+	if nsErr != nil {
+		fmt.Printf("Failed to create namespace: %s", nsErr)
+		skip = true
+		return
+	}
+
+	autherr = lc.Initialize(clientset, testingNamespace)
+	if autherr != nil {
+		skip = true
+		return
```

Comment:
You do an explanatory printf for the other skips, but not this one? Feels lonley